### PR TITLE
Refs #29548 -- Updated docs for MariaDB support.

### PR DESCRIPTION
**Pull request by felixxm**
Ticket [29548](https://code.djangoproject.com/ticket/29548). It's time officially supports MariaDB.

I updated docs with basic notes about MariaDB support.  

@adamchainz @orf Would you like to take a look? I'm not a MariaDB expert. Should we add some MariaDB caveats?

### CODE REVIEW

**Comment by orf** [comment]
I think this is all that is required right now. There are some issues I ran into regarding differing support for Window functions but that's out of the scope for these docs.

**Comment by felixxm** [comment]
@orf Thanks Tom. Can you say something more about falling tests? All tests pass on our CI, I also saw that the last build in django-docker-box was green for MariaDB 10.1-10.3 :thinking:

**Comment by orf** [comment]
It’s not failing tests, I was just remembering the work I did in https://github.com/django/django/pull/10228 - it seems MariaDB has diverged from MySQL a fair bit in its support for window functions. I was going to suggest adding a bit in the docs to mention that while MariaDB is based on MySQL the support for various things may differ significantly. 

Right now saying “it shares a backend with MySQL” may confuse people into thinking it’s a very similar database, which is not necessarily true.

File: docs/ref/databases.txt [hunk 1]
```diff
@@ -268,6 +268,18 @@ non-durable <https://www.postgresql.org/docs/current/non-durability.html>`_.
...
+.. _mariadb-notes:
+
+MariaDB notes
+=============
+
+.. versionadded:: 3.0
+
+Django supports MariaDB 10.1 and higher.
+
+MariaDB shares backend with MySQL, see the :ref:`MySQL notes <mysql-notes>` for
```
Comment (adamchainz) [inline docs/ref/databases.txt]: Maybe better wording: "To use MariaDB, use the MySQL backend, which is shared between the two. See the MySQL notes for more details."

Comment (felixxm) [reply docs/ref/databases.txt]: Updated.

**Pull request merged**